modules/social_features/social_profile/modules/social_profile_privacy/src/Service/SocialProfilePrivacyBatchHelper.php

```diff
@@ -0,0 +1,258 @@
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function getProfileName(ProfileInterface $profile = NULL) {
+    /** @var \Drupal\user\UserStorageInterface $user_storage */
+    $user_storage = $this->entityTypeManager->getStorage('user');
+
+    // Do nothing if no profile.
+    if ($profile == NULL) {
+      return '';
+    }
+
+    /** @var \Drupal\user\UserInterface $account */
+    $account = $user_storage->load($profile->getOwnerId());
+
+    // Set default profile name.
+    $account_name = $account->getAccountName();
+
+    // Get profile private fields list.
+    $uid = $account->id();
+    $private_fields_list = social_profile_privacy_private_fields_list($uid);
+
+    $account_name_fields = [
+      'field_profile_first_name',
+      'field_profile_last_name',
+    ];
+
+    foreach ($account_name_fields as $key => $account_name_field) {
+      if (in_array($account_name_field, $private_fields_list)) {
+        unset($account_name_fields[$key]);
+      }
+      else {
+        $account_name_fields[$key] = $profile->get($account_name_field)->getString();
+      }
+    }
+
+    if (!empty($account_name_fields) && !empty($full_name = implode(" ", $account_name_fields))) {
+      $account_name = $full_name;
+    }
+
+    if ($this->moduleHandler->moduleExists('social_profile_fields') && !in_array('field_profile_nick_name', $private_fields_list) && !empty($profile->get('field_profile_nick_name')->getString())) {
+      $account_name = $profile->get('field_profile_nick_name')->getString();
+    }
+
+    return $account_name;
+  }
+
+  /**
+   * Update profile names in a batch.
+   */
+  public static function bulkUpdateProfileNames() {
+    // Define batch process to update profile names.
+    $batch_builder = (new BatchBuilder())
+      ->setTitle(t('Updating profile names...'))
+      ->setFinishCallback([
+        SocialProfilePrivacyBatchHelper::class,
+        'finishProcess',
+      ])
+      ->addOperation([SocialProfilePrivacyBatchHelper::class, 'initOperation'], [
+        ['limit' => 50],
+      ])
+      ->addOperation([SocialProfilePrivacyBatchHelper::class, 'updateProcess'], [
+        ['limit' => 50],
+      ]);
+
+    batch_set($batch_builder->toArray());
+  }
+
+  /**
+   * Init operation task by retrieving all content to be updated.
+   *
+   * @param array $args
+   *   Arguments.
+   * @param array $context
+   *   An array that may or may not contain placeholder variables.
+   */
+  public static function initOperation(array $args, array &$context) {
+    // Init variables.
+    $limit = (int) $args['limit'];
+    $offset = (int) $context['sandbox'] ?? 0;
+
+    $userStorage = \Drupal::entityTypeManager()->getStorage('user');
+
+    $uids = $userStorage->getQuery()
+      ->condition('status', '1')
+      ->execute();
+    /** @var array $results */
+    $results = $userStorage->loadMultiple($uids);
```

Comment:
This can again lead to the loading of thousands of profiles at once, which is not good for a low PHP memory server, it would be better if we can pass on UIDs in the sandbox and load the user objects later which processing